cwltool/workflow.py

```diff
@@ -64,30 +65,36 @@ def match_types(sinktype, src, iid, inputobj, linkMerge, valueFrom):
         for st in src.parameter["type"]:
             srccopy = copy.deepcopy(src)
             srccopy.parameter["type"] = st
-            if not match_types(st, srccopy, iid, inputobj, linkMerge, valueFrom):
+            if not match_types(
+                    st, srccopy, iid, inputobj, linkMerge, valueFrom):
                 return False
         return True
     elif linkMerge:
-            if iid not in inputobj:
-                inputobj[iid] = []
-            if linkMerge == "merge_nested":
-                inputobj[iid].append(src.value)
-            elif linkMerge == "merge_flattened":
-                if isinstance(src.value, list):
-                    inputobj[iid].extend(src.value)
-                else:
-                    inputobj[iid].append(src.value)
+        if iid not in inputobj:
+            inputobj[iid] = []
+        if linkMerge == "merge_nested":
+            inputobj[iid].append(src.value)
+        elif linkMerge == "merge_flattened":
+            if isinstance(src.value, list):
+                inputobj[iid].extend(src.value)
             else:
-                raise WorkflowException("Unrecognized linkMerge enum '%s'" % linkMerge)
-            return True
-    elif valueFrom is not None or are_same_type(src.parameter["type"], sinktype) or sinktype == "Any":
-        # simply assign the value from state to input
-        inputobj[iid] = copy.deepcopy(src.value)
+                inputobj[iid].append(src.value)
+        else:
+            raise WorkflowException(
+                "Unrecognized linkMerge enum '%s'" % linkMerge)
```

Comment:
This block needs to be indented, part of the `elif linkMerge == "merge_flattened":` branch